stackrox/stackrox
# Delete obsolete processes reducers and sagas in UI

## Description

We are investigating **duplicate requests** because they increase risk of integration test failures (see manual testing below).

### Analysis

Risk container calls service functions directly instead of indirectly via reducers and sagas.

1. reducer actions
    * fetchProcesses: 3 calls of action in processSagas.js and 2 calls in Risk container of corresponding function from ProcessesService.js
    * fetchProcessesBaseline: 2 calls of action in processSagas.js
    * addDeleteProcesses: 1 call of action in processSagas.js and 2 calls in Risk container of corresponding function from ProcessService.js
    * lockUnlockProcesses: 1 call of action in processSagas.js and 2 calls in Risk container of corresponding function from ProcessService.js
2. reducer selectors
    * getProcessesByDeployment: 0 occurrences
    * getProcessesBaselineByDeployment: 0 occurrences
3. Risk container
    * mapDispatchToProps: 1 occurrence in CreatePolicyFromSearch which is unrelated to processes
    * mapStateToProps: 0 occurrences
    * useDispatch: 0 occurrences
    * useSelector: 0 occurrences

### Changed files

1. Edit reducers/index.js

2. Delete reducers/processes.js

3. Edit sagas/index.js

4. Delete sagas/processSagas.js

### Residue

1. Rewrite services/ProcessesService.js in TypeScript

## Checklist

## Testing Performed

### manual testing

1. `yarn build` in ui
2. `wc build/static/js/*.chunk.js` in ui
    * branch - master: -4300 = 10410170 - 10414470
3. `yarn start` in ui

    * Visit /main/risk: see requests from RiskTablePanel
        GET /v1/deploymentswithprocessinfo?pagination.offset=0&pagination.limit=50&pagination.sortOption.field=Deployment%20Risk%20Priority&pagination.sortOption.reversed=false&query=Orchestrator%20Component%3Afalse
        GET /v1/deploymentscount?query=Orchestrator%20Component%3Afalse
    
    * Click a deployment: see **duplicate requests** before changes but not after changes
        GET /v1/processes/deployment/id/grouped/container from processSagas before changes but not after changes
        GET /v1/deploymentswithrisk/id from RiskSidePanel
        GET /v1/processes/deployment/id/grouped/container from RiskSidePanel

### integration testing

1. `yarn cypress-open` in ui/apps/platform: some tests failed in local deployment because Collector Status Unhealthy
    * risk/deploymentTimeline.test.js
    * risk/podTimeline.test.js
    * risk/processtags.test.js
    * risk/risk.test.js
    * risk/riskSearchToPolicy.test.js
    * risk/timelineOverview.test.js
